src/averageAges.js

```diff
@@ -56,6 +76,24 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const children = people.filter(child =>
+    onlyWithSon
+      ? getChildren(people, child)
+        && child.sex === 'm'
+      : getChildren(people, child));
+
+  return children.reduce((sum, child) =>
+    sum + getDiffAge(children, people, child), 0) / children.length;
+}
+
+function getChildren(people, child) {
+  return people.find(mother => mother.name === child.mother);
+}
+
+function getDiffAge(children, people, child) {
+  const motherBirth = people.find(mother => mother.name === child.mother);
+
+  return child.born - motherBirth.born;
```

Comment:
You better do it inside reduce method in the return statement. Now it just overcomplicates your code.